recirq/toric_code/toric_code_rectangle.py

```diff
@@ -0,0 +1,283 @@
+
+      c0↓  c1↓  c2↓  c3↓  c4↓
+    r0→      Q    C
+
+    r1→ Q    C    Q    C
+
+    r2→ Q    Q    C    Q    C
+
+    r3→      Q    Q    C    Q
+
+    r4→           Q    Q
+
+    In this example, the "captain" qubits are (0, 2), (1, 1), (1, 3), (2, 2), (2, 4), and (3, 3).
+    We define the orientation by the "row vector" from the (row=0, col=0) "captain" qubit (0, 2)
+    to the (row=1, col=0) "captain" qubit (1, 1) in GridQubit coordinates, which is (1, -1) here.
+
+    In particular, this would be represented by
+    ToricCodeRectangle(
+        origin_qubit=cirq.GridQubit(0, 2),
+        row_vector=(1, -1),
+        rows=2,
+        cols=3,
+    )
+    """
+
+    def __init__(
+        self,
+        origin_qubit: cirq.GridQubit,
+        row_vector: Tuple[int, int],
+        rows: int,
+        cols: int,
+    ):
+        """
+
+        Args:
+            origin_qubit: "Captain" (top) qubit of the upper-left X plaquette, in the toric code
+                reference frame
+            row_vector: Displacement between origin_qubit and the opposite qubit in its X plaquette,
+                which is the leftmost "captain" qubit in the second row of X plaquettes. This
+                determines the orientation of the toric code on the qubit grid. Must be (±1, ±1).
+            rows: Number of rows of X plaquettes
+            cols: Number of columns of X plaquettes
+        """
+        if np.any(np.abs(row_vector) != np.array([1, 1])):
+            raise ValueError(f"Illegal {row_vector=}: must be (±1, ±1)")
+
+        if rows < 1 or cols < 1:
+            raise ValueError(f"rows ({rows}) and cols ({cols}) must both be at least 1")
+
+        self.origin_qubit = origin_qubit
+        self.row_vector = np.array(row_vector, dtype=int)
+        self.col_vector = np.array((-row_vector[1], row_vector[0]), dtype=int)
+        self.rows = rows
+        self.cols = cols
+        self.qubits_reading_order = self._qubits_reading_order()
+
+    def __repr__(self) -> str:
+        return (
+            f"ToricCodeRectangle(origin_qubit=cirq.GridQubit{self.origin_qubit}, "
+            f"row_vector={tuple(self.row_vector)}, rows={self.rows}, cols={self.cols})"
+        )
+
+    @property
+    def captain_qubits(self) -> Set[cirq.GridQubit]:
+        return {
+            self.x_plaquette_to_captain(row, col)
+            for row, col in self.x_plaquette_indices()
+        }
+
+    @property
+    def qubits(self) -> Set[cirq.GridQubit]:
+        return {
+            qubit
+            for row, col in self.x_plaquette_indices()
+            for qubit in self.x_plaquette_to_qubits(row, col)
+        }
+
+    def _qubits_reading_order(self) -> Tuple[cirq.GridQubit, ...]:
+        """Qubits listed in "reading order."
+
+        The qubits are enumerated starting with the "captain" of x_plaquette (0, 0)
+        being map[1] and then reading right (the first self.cols qubits are the captains
+        of row 0) and going down row by row.
```

Comment:
Update docstring to reflect the tuple and zero-indexing (it looks like the old dict and one-indexing docstring)

Reply:
Done.